[BE]Clean up old APIs in AOTI c shim

yiming0416 opened the pull request:
Summary:
The shims for aten ops are now generated by torchgen. But there are some still old APIs in `aoti_torch/c/shim.h`

This diff moves the old to-be-deprecated APIs for aten ops to a separate header file `shim_deprecated.h`

The to-be-deprecated APIs are determined by comparing APIs in `shim.h` and ops in `fallback_ops.py`

Test Plan:
CI

Rollback Plan:

Differential Revision: D78378373

desertfire commented on torch/csrc/inductor/aoti_torch/c/shim_deprecated.h:
```diff
@@ -0,0 +1,151 @@
+// torchgen and locate in torch/csrc/inductor/aoti_torch/generated
+
+#ifdef __GNUC__
+#define AOTI_TORCH_EXPORT __attribute__((__visibility__("default")))
```
This and the following macros will cause duplicated macro definition, which is not recommended. You can split them into a separate macros. h header file.

desertfire commented on torch/csrc/inductor/aoti_torch/c/shim_deprecated.h:
```diff
@@ -0,0 +1,151 @@
+#define AOTI_TORCH_SUCCESS 0
+#define AOTI_TORCH_FAILURE 1
+
+AOTI_TORCH_EXPORT AOTITorchError aoti_torch__embedding_bag(
```
Please also mark these functions as deprecated similar to here, https://github.com/pytorch/pytorch/blob/bc9091a524a1ebe4de16af4dd8f442db7d1cb138/c10/core/ScalarType.h#L378

desertfire commented on torch/csrc/inductor/aoti_torch/c/shim_deprecated.h:
```diff
@@ -0,0 +1,151 @@
+    AtenTensorHandle* ret // returns new reference
+);
+
+#ifdef __cplusplus
```
How about `_scaled_dot_product_flash_attention` and `_scaled_dot_product_efficient_attention`?

yiming0416 replied:
@desertfire added them

desertfire commented on torch/csrc/inductor/aoti_torch/c/shim.h:
```diff
@@ -349,51 +299,6 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch_create_tensor_from_blob_v2(
-);
-
 AOTI_TORCH_EXPORT AOTITorchError
 aoti_torch__scaled_dot_product_flash_attention_v2(
```
This one needs to be moved as well.

yiming0416 replied:
@desertfire  I can move it to `shim_deprecated.h` but I cannot mark it with `[[deprecated]]`. Otherwise build failed as in `shim_common.cpp` `aoti_torch__scaled_dot_product_flash_attention_v2` is being called inside `aoti_torch__scaled_dot_product_flash_attention` and all warnings being treated as errors in CI.

yiming0416 committed:
[BE]Clean up old APIs in AOTI c shim (#158400)

Summary:

The shims for aten ops are now generated by torchgen. But there are some still old APIs in `aoti_torch/c/shim.h`

This diff moves the old to-be-deprecated APIs for aten ops to a separate header file `shim_deprecated.h`

The to-be-deprecated APIs are determined by comparing APIs in `shim.h` and ops in `fallback_ops.py`

Test Plan:
CI

Rollback Plan:

Reviewed By: desertfire, jingsh

Differential Revision: D78378373
